uniffi/src/lib.rs

```diff
@@ -2,352 +2,293 @@
-//! ```
-//!
-//! This would output a rust file named `example.uniffi.rs`, ready to be
-//! included into the code of your rust crate like this:
-//!
-//! ```text
-//! include!(concat!(env!("OUT_DIR"), "/example.uniffi.rs"));
-//! ```
-//!
-//! ### 4) Build your crate as a `cdylib`
-//!
-//! The generated code automatically declares the necessary `pub extern "C"` functions and type
-//! conversions to expose your rust code over a C-compatible foreign function interface.
-//! Make it available for use by building it as a `cdylib` whose name is prefixed with the
-//! string "uniffi_". For this example component, the necessary config in `Cargo.toml` would
-//! be:
-//!
-//! ```text
-//! [lib]
-//! crate-type = ["cdylib"]
-//! name = "uniffi_example"
-//! ```
-//!
-//! Running `cargo build` on your crate should produce an appropriate shared library for your
-//! system, in this case something like `libuniffi_example.so` or `libuniff_example.dylib`.
-//! In addition to the compiled Rust code, this shared library will contain a copy of the abstract
-//! interface definitions from your IDL file, allowing it to be safely consumed from other
-//! languages.
-//!
-//! ### 5) Generate foreign language bindings for the library
-//!
-//! The `uniffi` crate provides a command-line tool that can take the dynamic library built
-//! from a rust component, extract the embedded IDL details, and produce produce code for
-//! consuming that library in any of several supported languages.
-//!
-//! To help ensure this is done with appropriate config for your component, add a `src/main.rs`
-//! script to call `uniffi::run_bindgen_for_component`, like this:
-//!
-//! ```text
-//! fn main() {
-//!    uniffi::run_bindgen_for_component("example").unwrap();
-//!}
-//! ```
-//!
-//! Then you can generate bindings using `cargo run generate` in your crate. For example, to
-//! generate python bindings for the example component, run:
-//!
-//! ```text
-//! cargo run generate -l python
-//! ```
-//!
-//! This will produce a file `example.py` in the cargo target directory, containing python bindings
-//! to load and use the compiled rust code via its C-compatible FFI.
-//!
-
-use std::convert::TryInto;
-use std::io::prelude::*;
-use std::{
-    env,
-    fs::File,
-    path::{Path, PathBuf},
-};
-
-use anyhow::anyhow;
-use anyhow::bail;
-use anyhow::Result;
-
-pub mod bindings;
-pub mod interface;
-pub mod scaffolding;
-pub mod support;
-
-use scaffolding::RustScaffolding;
-
-pub(crate) fn slurp_file(file_name: &str) -> Result<String> {
-    let mut contents = String::new();
-    let mut f = File::open(file_name)?;
-    f.read_to_string(&mut contents)?;
-    Ok(contents)
+use anyhow::{bail, Result};
+use bytes::buf::{Buf, BufMut};
+use ffi_support::ByteBuffer;
+use std::convert::TryFrom;
+
+// I'd be nice this module was behind a cfg(test) guard, but it doesn't
+// work between crates so let's hope LLVM tree-shaking works well.
+pub mod testing;
```

Comment:
I guess we could consider moving this into a fourth "dev-dependencies" crate, but I think this will be fine as is, and I have confidence in the tree-shaking stuff 🤞